server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1886,6 +1886,10 @@ public Engine.Searcher acquireSearcher(String source) {
     }
 
     private void markSearcherAccessed() {
+        if (isSearchIdle()) {
```

Comment:
The `isSearchIdle` condition isn't sufficient, I think.

For example, if an explicit refresh interval is specified, shards will never go idle, but I believe `isSearchIdle` will still return `true` (because it's just checking the time).

I'm not sure what the best solution is. The ideas I have are:
1. Copy more of the conditions from `scheduleRefresh` (i.e. ` refreshListeners.refreshNeeded() == false && isSearchIdleSupported() && isSearchIdle() && indexSettings.isExplicitRefresh() == false && active.get()`).
2. Move the counter into the `location != null` block of `awaitShardSearchActive` (which I *think* is the code path for the blocking call for waking from idle).

Reply:
I put the counter in `location != null `block and tested it with refresh interval explicitly set to make sure it's the right behavior